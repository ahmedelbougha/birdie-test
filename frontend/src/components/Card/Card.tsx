import { useContext } from 'react';
import { ThemeContext } from 'styled-components';
import { CardProps, StyledCard, StyledCardDivider } from './';

/**
 * Card component for the care recipient to be displayed in the home page
 * @param CardProps
 * @returns JSX.Element
 */
export function Card({
  title,
  image,
  children,
}: CardProps): JSX.Element {
  const themeContext = useContext(ThemeContext);
  return (
    <StyledCard>
      <div>
        <img src={image || themeContext.placeHolderImage} alt={title} />
      </div>
      <div>
        <h2>{title}</h2>
        <StyledCardDivider />
        {children}
      </div>
    </StyledCard>
  );
}
