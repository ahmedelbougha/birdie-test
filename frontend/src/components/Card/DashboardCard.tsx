import { useContext } from 'react';
import { ThemeContext } from 'styled-components';
import { CardProps, StyledDashboardCard } from '.';

export default function DashboardCard({ title, image }: CardProps) {
  const themeContext = useContext(ThemeContext);
  return (
    <StyledDashboardCard>
      <div>
        <img src={image || themeContext.placeHolderImage} alt={title} />
      </div>
      <div>
        <h2>{title}</h2>
      </div>
    </StyledDashboardCard>
  );
}
