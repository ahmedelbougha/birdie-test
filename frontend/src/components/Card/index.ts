export { Card as default } from "./Card";
export { type CardProps } from "./Card.d";
export {
    StyledCard,
    StyledCardDivider,
    StyledCardsWrapper as CardsWrapper,
    StyledDashboardCard
} from "./Card.styled";
export { default as DashboardCard } from "./DashboardCard";

