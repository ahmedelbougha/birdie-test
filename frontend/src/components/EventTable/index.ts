export { EventTable as default } from "./EventTable";
export { type TableProps } from "./EventTable.d";
export { StyledTable } from "./Table.styled";

