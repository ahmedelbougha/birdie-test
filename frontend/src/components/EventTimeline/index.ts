export { EventTimeline as default } from "./EventTimeline";
export { type EventTimelineElementProps, type EventTimelineProps } from "./EventTimeline.d";
export { default as EventTimelineElement } from "./EventTimelineElement";

