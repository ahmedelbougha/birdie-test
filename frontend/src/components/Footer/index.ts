export { Footer as default } from "./Footer";
export {
    StyledFooter,
    StyledFooterLogo
} from "./Footer.styled";

