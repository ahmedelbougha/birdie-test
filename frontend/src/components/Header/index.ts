export { Header as default } from "./Header";
export {
    StyledHeader,
    StyledHeaderLogo,
    StyledNav
} from "./Header.styled";

