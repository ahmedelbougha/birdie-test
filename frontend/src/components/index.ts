export { CardsWrapper, DashboardCard, default as Card } from "./Card";
export { default as EventTable } from "./EventTable";
export { default as EventTimeline, EventTimelineElement } from "./EventTimeline";
export { default as Footer } from "./Footer";
export { default as Header } from "./Header";
export { default as Spinner } from "./Spinner";
export { Button, Container, GlobalStyles, Layout, StyledError } from "./styles";

