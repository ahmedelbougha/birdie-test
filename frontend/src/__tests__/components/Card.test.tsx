import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { Button } from "../../components";
import Card, { CardProps } from "../../components/Card";
import { theme } from "../../utils/constants";

const cardProps: CardProps = {
  title: "Test Card",
  image: "",
  children: (
    <>
      <p>Birdie Client</p>
      <Button> Details </Button>
    </>
  ),
};

describe("Card", () => {
  it("renders", () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Card {...cardProps} />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
