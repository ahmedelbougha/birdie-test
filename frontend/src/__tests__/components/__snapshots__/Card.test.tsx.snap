// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders 1`] = `
<div
  className="sc-gswNZR eVCMhJ"
>
  <div>
    <img
      alt="Test Card"
      src="placeholder.png"
    />
  </div>
  <div>
    <h2>
      Test Card
    </h2>
    <div
      className="sc-bcXHqe AOYhz"
    />
    <p>
      Birdie Client
    </p>
    <button
      className="sc-kDvujY fuLqXM"
    >
       Details 
    </button>
  </div>
</div>
`;
